src/schema-processor/json/parser.ts

```diff
@@ -223,4 +223,69 @@ export class Parser {
         throw new Error(`field ${field} not specified in serialization info`);
     }
   }
+
+  /**
+   * ExtractMetadata return metadata from JSON schema
+   *
+   * @param {string | JSON} schema - JSON schema
+   * @returns SchemaMetadata
+   */
+  public static extractMetadata(schema: string | JSON): SchemaMetadata {
+    const parsedSchema = typeof schema === 'string' ? JSON.parse(schema) : schema;
+    const md = parsedSchema.$metadata;
+    if (!md) {
+      throw new Error('$metadata is not set');
+    }
+    return md;
+  }
+
+  /**
+   * ExtractCredentialSubjectTypes return credential subject types from JSON schema
+   *
+   * @param {string | JSON} schema - JSON schema
+   * @returns `Promise<Array<string>>`
+   */
+  public static async extractCredentialSubjectTypes(schema: string): Promise<Array<string>> {
```

Comment:
id doesn't extract  `CredentialSubjectTypes`, it extracts 'properties', are you sure this method has to be public ?

Reply:
Yes. This method helps get data from schema.